tracer/src/Datadog.Trace/ClrProfiler/ServerlessInstrumentation/AWS/LambdaCommon.cs

```diff
@@ -206,5 +208,26 @@ private static void WriteRequestHeaders(WebRequest request, IDictionary<string,
             var proxyInstance = obj.DuckAs<ILambdaContext>();
             return proxyInstance?.ClientContext?.Custom;
         }
+
+        internal static void InjectTraceContext(object executionContext, string traceId, string parentId, string samplingPriority)
+        {
+            var clientContextString = $"{{ \"custom\": {{ \"x-datadog-trace-id\": \"{traceId}\", \"x-datadog-parent-id\": \"{parentId}\", \"x-datadog-sampling-priority\": \"{samplingPriority}\" }}}}";
```

Comment:
Related to https://github.com/DataDog/dd-trace-dotnet/pull/2693/files#r855471313

I looked around the codebase and I found another bit of code that re-uses our main context propagation infra so you don't have to roll your own inside `LambdaCommon.InjectTraceContext`: https://github.com/DataDog/dd-trace-dotnet/blob/540b5b98797ae5dc37bdf485200c9ea3853fa446/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/SQS/ContextPropagation.cs

```csharp
        private static void Inject<TMessageRequest>(SpanContext context, IDictionary messageAttributes)
        {
            // Consolidate headers into one JSON object with <header_name>:<value>
            var sb = Util.StringBuilderCache.Acquire(Util.StringBuilderCache.MaxBuilderSize);
            sb.Append('{');
            SpanContextPropagator.Instance.Inject(context, sb, default(StringBuilderCarrierSetter));
            sb.Remove(startIndex: sb.Length - 1, length: 1); // Remove trailing comma
            sb.Append('}');


            var resultString = Util.StringBuilderCache.GetStringAndRelease(sb);
            messageAttributes[SqsKey] = CachedMessageHeadersHelper<TMessageRequest>.CreateMessageAttributeValue(resultString);
        }

        private readonly struct StringBuilderCarrierSetter : ICarrierSetter<StringBuilder>
        {
            public void Set(StringBuilder carrier, string key, string value)
            {
                carrier.AppendFormat("\"{0}\":\"{1}\",", key, value);
            }
        }
```

You could use the same logic that's in the `Inject` method but modify the `StringBuilder` so it starts with `"{ custom: {"` and ends with `"} }"`. And then once you get the `resultString`, you use that as the `clientContextString` that's the basis for the `OriginalRequest` updates